### ISSUE TITLE
Need to limit the chart types depending on the report

### ISSUE THREAD

**Issue by matmaxgeds** [somaliaims/Somali-AIMSUI#364]
E.g. for the sectors, there is no point to have a line chart as it doesn't make sense......I will have to fill in a list here, but only once we have finalised the bits with planned/actual disbursements....and removing 'number of projects'

**Comment by raashidahmad** [comment]
@matmaxgeds we have removed chart types as with stacked bar chart data, what is to do with this task?

**Comment by matmaxgeds** [comment]
writing this offline - will be entered in 10 mins

**Comment by raashidahmad** [comment]
Fixed.

**Comment by matmaxgeds** [comment]
Hi @raashidahmad, just one thing missed here:

- [ ] Please make the table for the sector and location reports like the 'Budget Report' table, where the sub-totals are in the columns, not encoded in the title of the parent row

**Comment by raashidahmad** [comment]
Fixed.

**Comment by matmaxgeds** [comment]
@raashidahmad great - just the 'planned disbursements' column is not subtotalling like the other ones

**Comment by raashidahmad** [comment]
@matmaxgeds Thanks. Implemented as suggested.

**Comment by matmaxgeds** [comment]
@raashidahmad 

See above: "Budget report:
Table main row: sectors or location (user to select one, the one that is not selected, becomes the sub-row
Table sub-row: sectors or location (see above)"

Let me know if I should do a mockup to make this clearer?

**Comment by raashidahmad** [comment]
@matmaxgeds Yes would need mockup to understand this idea.

**Comment by matmaxgeds** [comment]
@raashidahmad - here is one, it is a bit different to the other reports, but with no filters so hopefully simpler.
[Budget report mockup.xlsx](https://github.com/somaliaims/Somali-AIMSUI/files/4174162/Budget.report.mockup.xlsx)

**Comment by raashidahmad** [comment]
@matmaxgeds thanks for the mockup. I can see and understand the idea, will require to re-write the API, and then accordingly adjust the UI. Do i need to take priority on this one or just with the normal priority i can do it?

**Comment by matmaxgeds** [comment]
Good question, I think normal priority is fine

**Comment by raashidahmad** [comment]
@matmaxgeds while trying to formulate the report provided in the mockup, i am unclear how to formulate it, for example, please see the questions below:
1. When group by Sectors, a list of locations will be listed under each sector.
2. We know that we have different percentage of the disbursements for sector and locations under an year.
3. How to sum up the values for year, then how to allocate those values as a total and then for each sector/location under an year.

**Comment by matmaxgeds** [comment]
@raashidahmad 
1. Correct - and no individual projects
2. Correct - so the disbursement per sector, per location per year = the sector % * location % * disbursement for that year
3. Sum these up for all the projects to get the total for each year, for each sector/location combination.

Does that help?

**Comment by raashidahmad** [comment]
@matmaxgeds thanks, yes helpful, working on it, will get back to you if stuck somewhere.

**Comment by raashidahmad** [comment]
@matmaxgeds upon further analysis when writing the algorithm for report, please see the findings below. 
The percentage to every sector or location is assigned at project level. A sector or location may appear several times under an year with different percentages for each project.
For example, a Sector may have a percentage of 10% for one project, while 20% for another project. Now look at the arrangement of the report, In this case consider we have grouped the report by sector:

Health (A sector, we may sum it as "Sum all the percentage values for all the projects where it occurs in the particular year)

	1. Somaliland (Very similar, Sum all the percentage values for projects in an year)
        2. Other locations

By looking at the arrangement given above, the sum of financial value for the sector "Health" is sum of all the percentages occurring in different projects and not a single percentage for a financial year. The issue arise here, that how to get the percentage of this sector so this may be applied all the locations under it organized in hierarchical manner. Similarly, please think about what if we change the arrangement and make it group by locations.

**Comment by matmaxgeds** [comment]
@raashidahmad  - maybe it is easier to think of it in terms of money, not in terms of percentages as you just use the percentages to calculate the money, and then add the money up, not try to add the percentages up.. E.g. Project A has 20% for health, and 60% for Somaliland, 40% for Jubbaland, and a 2019 disbursement of 100,000. So the table figure for health in Somaliland = 0.2*0.6*100000, and health in Jubbaland is 0.2*0.4*100,000.

The value in the table for health is the sum of all the different location amounts for health - I think the table will add those up for you?

**Comment by raashidahmad** [comment]
@matmaxgeds correct, this is any easy way to think about and makes sense. We will calculate the values for each location and sector and then will add the values under a sector or location. Thanks for the help, working on it.

**Comment by raashidahmad** [comment]
@matmaxgeds closing this issue, as we already discussed reports on email. We may work on budget report and create a new issue for that after you do some paper work on it. 
Please ignore the similar comment i posted mistakenly against some other ticket and now cannot find where did i post.
